src/GraphQL/Utilities/SchemaBuilder.cs

```diff
@@ -268,7 +268,9 @@ protected virtual IObjectGraphType ToObjectGraphType(GraphQLObjectTypeDefinition
 
             var type = _types.TryGetValue(name, out var t)
                 ? t as ObjectGraphType ?? throw new InvalidOperationException($"Type '{name} should be ObjectGraphType")
-                : new ObjectGraphType { Name = name };
+                : ServiceProvider.GetRequiredService<ObjectGraphType>();
```

Comment:
```suggestion
                : ServiceProvider.GetService<ObjectGraphType>() ?? new ObjectGraphType();
```
Probably need to add the necessary extension method within `Utilities\ServiceProviderExtensions.cs`